Move plone.app.caching documentation to deployment/caching (#1449)
plone/documentation: docs/deployment/caching/index.md

---
myst:
  html_meta:
    "description": "Caching Plone"
    "property=og:description": "Caching Plone"
    "property=og:title": "Caching Plone"
    "keywords": "Plone, deployment, automation, caching"
---

(caching-label)=

# Caching

## Overview

> HTTP caching is a technique used to speed up the delivery of web content by storing previously requested resources (such as images, scripts, and stylesheets) in a cache. When a user requests a resource, the cache can serve the resource directly from its storage, rather than having to fetch it from the original source.

Reviewer: ```suggestion
HTTP caching is a technique used to speed up the delivery of web content by storing previously requested resources (such as images, scripts, and stylesheets) in a cache. 
When a user requests a resource, the cache can serve the resource directly from its storage, rather than having to fetch it from the original source.
```

> It is possible to use a web accelerator, like {term}`Varnish` or {term}`Squid`, to implement HTTP caching in your own premises. The usual setup will have the web accelerator installed between Plone and the Internet, and when a request is made to the site the web accelerator will intercept the request and check to see if it has a cached copy of the requested resource. If a cached copy is found, the accelerator will serve the cached copy directly to the client, otherwise it will make a request to the backend Plone server and then store a copy on the content in its local cache.

Reviewer: Do we really want to mention Squid these days? 
Also here: split to one sentence per line
```suggestion
It is possible to use a web accelerator, like {term}`Varnish` to implement HTTP caching in your own premises. 
The usual setup will have the web accelerator installed between Plone and the Internet, and when a request is made to the site the web accelerator will intercept the request and check to see if it has a cached copy of the requested resource. 
If a cached copy is found, the accelerator will serve the cached copy directly to the client, otherwise it will make a request to the backend Plone server and then store a copy on the content in its local cache.
```

> A Content Delivery Network (CDN) can also be used to implement the HTTP caching and speed up the delivery of web content. A CDN is a network of servers located in multiple geographic regions that work together to deliver web content to users more quickly and efficiently. When a user requests a resource from a website that is using a CDN, the request is routed to the nearest server in the CDN, rather than having to travel all the way to the website's origin server.

Reviewer: ```suggestion
HTTP caching and the speed of delivering web content can be improved using a Content Delivery Network (CDN).
A CDN is a network of servers located in various geographic regions that work together to deliver web content to users quickly and efficiently. 
When a user requests a resource from a website that uses a CDN, the request is directed to the closest server in the CDN, rather than having to travel all the way to the website's origin server.
```

## Plone support

> Plone, since version 4.1, ships with a powerful and extensible `HTTP` and `In-Memory` cache support, implemented by the package `plone.app.caching`.

Reviewer: I do not think we need a history lesson here ;)
```suggestion
Plone ships with a powerful and extensible `HTTP` and `In-Memory` cache support, implemented by the package `plone.app.caching`.
```
